include/avif/avif.h

```diff
@@ -749,6 +749,13 @@ AVIF_API void avifImageStealPlanes(avifImage * dstImage, avifImage * srcImage, a
 // the number of threads that can ever be in flight at a given time, please account for this
 // accordingly.
 
+// ---------------------------------------------------------------------------
+// Scaling
+
+// Scales the YUV/A planes in-place. dstWidth and dstHeight must both be <= AVIF_DEFAULT_IMAGE_DIMENSION_LIMIT and
+// dstWidth*dstHeight should be <= AVIF_DEFAULT_IMAGE_SIZE_LIMIT.
+avifBool avifImageScale(avifImage * image, uint32_t dstWidth, uint32_t dstHeight, avifDiagnostics * diag);
```

Comment:
Please look into whether the public function needs a `diag` parameter. I defer this decision to you.

Reply:
I think it doesn't hurt to have it, it's always nice to have more detailed error messages than just a status, or "false". Maybe clarify that it's optional (can be null).

Reply:
i saw that a few other public non encoder/decoder functions also have to diag parameter. so i thought i was okay to keep.

have clarified that it can be null.